libcxx/test/std/iterators/iterator.primitives/range.iter.ops/range.iter.ops.advance/iterator_count_sentinel.pass.cpp

```diff
@@ -42,6 +42,13 @@ constexpr void check_forward(int* first, int* last, std::iter_difference_t<It> n
     // regardless of the iterator category.
     assert(it.stride_count() == M);
     assert(it.stride_displacement() == M);
+    if (n == 0) {
+      assert(it.equals_count() == 0);
+    } else {
+      assert(it.equals_count() > 0);
+      assert(it.equals_count() == M || it.equals_count() == M + 1);
```

Comment:
Question: why are there two possibilities?

Reply:
The M case is the "normal" case where the iterator is incremented without the bound being "hit" (it may equal the bound though, but this last equals check shouldn't be executed).

The "M + 1" case happens when the advancing `while` loop is exited because of the bounds check, so the bound is "hit". Example: Calling `std::ranges::advance(it, 999999, s)` on a range `{ 42, 3, 4 }` (size 3) will do 4 (3 + 1) bounds checks.

Reply:
Thanks for explaining! In that case, we should either pass the expected number of comparisons as a parameter to this function, or pass a boolean parameter to indicate whether we expect the algorithm to hit the sentinel or not. Test cases should be simple so that we can be sure exactly what we're testing.

Reply:
That makes perfect sense. I'll update the test accordingly.

Reply:
done!